lib/marklogic.js

```diff
@@ -741,17 +741,23 @@ function initClient(client, inputParams) {
       connectionParams.host = 'localhost';
     } else if (key === 'port') {
       connectionParams.port = 8000;
-    } else if (key === 'authType') {
-      connectionParams.authType = 'DIGEST';
     }
   }
+  if(!connectionParams.apikey && !connectionParams.authType){
+    connectionParams.authType = 'DIGEST';
+  }
+  if(connectionParams.apikey){
+    connectionParams.host = connectionParams.tenancyUrl;
+    connectionParams.port = 443;
+  }
   connectionParams.lookup = prefLookup;
 
-  const authType = connectionParams.authType.toUpperCase();
+  const authType = connectionParams.authType? connectionParams.authType.toUpperCase():null;
   if (authType !== 'NONE' &&
```

Comment:
Shouldn't this really say "If authType is digest or basic, and either user or password is null, then throw an error"? 

I ask because I got this error when I did "apiKey" instead of "apikey". And it's confusing because the error is assuming that I'm trying to do digest or basic authentication.

Reply:
I see "CLOUD" is added to the list, but I think the code would be simpler and more future-proof if it instead said "If authType is digest or basic" as opposed to maintaining a list of all the other auth types.